### PR TITLE
Implement validation to detect when PCB components are placed outside board boundaries

### PR DESCRIPTION
**Pull request by AnasSarkiz**
**Features:**
- Validates component placement against board boundaries (rectangular and custom outlines)
- Rotation-aware bounding box calculations for accurate collision detection
- Detailed error messages with component names and overlap distances
- Integrates with existing DRC pipeline in Board component

**Implementation:**
- [validatePcbComponentsWithinBoardBounds()](cci:1://file:///Users/anassarkiz/ts/core/lib/utils/pcb-validation/validatePcbComponentsWithinBoardBounds.ts:128:0-184:1) - Core validation logic
- [PcbComponentOutsideBoardErrorData](cci:2://file:///Users/anassarkiz/ts/core/lib/errors/PcbComponentOutsideBoardError.ts:0:0-16:1) - Error interface matching circuit-json schema
- Component bounds calculation with rotation matrix transformation
- Board bounds calculation supporting both width/height and outline definitions

**Integration:**
- Runs during [doInitialPcbDesignRuleChecks()](cci:1://file:///Users/anassarkiz/ts/core/lib/components/normal-components/Board.ts:241:2-246:3) phase (early validation, routing-independent)
- Uses `pcb_component_outside_board_error` database table
- Provides helpful error messages like "Component R1 extends outside board boundaries by 3.78mm"

**Tests:**
- Comprehensive test suite covering rectangular boards, custom outlines, rotation, and multi-component scenarios
- All edge cases validated including components within bounds (no false positives)

Closes: [Issue #1323 ]
/claim #1323

### CODE REVIEW

**Review by seveibar — COMMENTED** [review]
Sorry, this should be implemented in tscircuit/checks with the other drc checks. Try to use functions from tscircuit/math-utils where applicable

File: lib/utils/pcb-validation/validatePcbComponentsWithinBoardBounds.ts [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+function calculateOverlapDistance(
+  componentBounds: ComponentBounds,
+  boardBounds: BoardBounds,
+): number {
+  const leftOverlap = Math.max(0, boardBounds.min_x - componentBounds.min_x)
+  const rightOverlap = Math.max(0, componentBounds.max_x - boardBounds.max_x)
+  const topOverlap = Math.max(0, boardBounds.min_y - componentBounds.min_y)
+  const bottomOverlap = Math.max(0, componentBounds.max_y - boardBounds.max_y)
+
+  return Math.max(leftOverlap, rightOverlap, topOverlap, bottomOverlap)
```
Comment (seveibar) [inline lib/utils/pcb-validation/validatePcbComponentsWithinBoardBounds.ts:126]: Try to use math-utils

**Comment by AnasSarkiz** [comment]
> Sorry, this should be implemented in tscircuit/checks with the other drc checks. Try to use functions from tscircuit/math-utils where applicable

Noted. This will be addressed tomorrow. It's too late now

**Comment by imrishabh18** [comment]
@AnasSarkiz Yeah, I guess you can add the rotation handling logic and the board outline conditions
